Web Inspector: InspectorFrontendHost should not call setPageAndTextZoomFactors on the LocalFrame

nipunshukla21 opened the pull request:
#### 795dbd901705376617ab877a079d2e6ca715974d
<pre>
Web Inspector: InspectorFrontendHost should not call setPageAndTextZoomFactors on the LocalFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=280493">https://bugs.webkit.org/show_bug.cgi?id=280493</a>
<a href="https://rdar.apple.com/136798696">rdar://136798696</a>

Reviewed by BJ Burg.

Setting zoom for inspector previously used a function which recursively iterates through
the frame stack and sets state. However, we were using a variant which would not zoom
into RemoteFrames. This patch changes that to use the function from the UI Process,
which can traverse the full frame tree including local and remote frames.

* Source/WebCore/inspector/InspectorFrontendClient.h:
(WebCore::InspectorFrontendClient::setPageAndTextZoomFactors):
(WebCore::InspectorFrontendClient::pageZoomFactor const):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::setZoomFactor):
(WebCore::InspectorFrontendHost::zoomFactor):
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.cpp:
(WebKit::RemoteWebInspectorUIProxy::setPageAndTextZoomFactors):
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.h:
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.messages.in:
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.cpp:
(WebKit::WebInspectorUIProxy::setPageAndTextZoomFactors):
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.h:
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.messages.in:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::setPageAndTextZoomFactors):
(WebKit::RemoteWebInspectorUI::pageZoomFactor const):
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::setPageAndTextZoomFactors):
(WebKit::WebInspectorUI::pageZoomFactor const):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.h:

Canonical link: <a href="https://commits.webkit.org/301271@main">https://commits.webkit.org/301271@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7daa295cc441c06c8eb563ad558434472cbff5ed

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124988 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44657 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35394 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131837 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76944 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126865 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45349 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53223 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95129 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63170 "Found 60 new test failures: compositing/clipping/border-radius-with-composited-descendant-nested.html compositing/clipping/border-radius-with-composited-descendant.html compositing/overlap-blending/children-opacity-huge.html compositing/overlap-blending/children-opacity-no-overlap.html compositing/overlap-blending/nested-non-overlap-clipping.html compositing/overlap-blending/nested-overlap.html compositing/transforms/clipping-layer-and-flattening-layer.html compositing/transforms/perspective-transform-box.html compositing/transforms/perspective-with-clipping.html compositing/webgl/update-composited-canvas-layer.html ... (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127942 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36191 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111782 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75676 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35118 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29934 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75318 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105954 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30165 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134511 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51811 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39612 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103605 "13 flakes 57 failures") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52236 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107994 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103381 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26420 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48724 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27012 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/48837 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51695 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57490 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51069 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54426 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52761 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8500ccdae3e8fb2f6d948702eb3017f9dc8213c5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50450

nipunshukla21 commented:
Sorry for the extended delay in updating this PR. I've changed it based on your feedback but it's still missing tests and I'm unable to manually verify that this change works as the only [example of this](https://developer.apple.com/documentation/safariservices/creating-safari-web-inspector-extensions?language=objc) I could find crashes upon switching to the inspector tab, both with and without site isolation and with and without my changes. I did modify the manifest to include what is shown [here](https://developer.apple.com/documentation/safariservices/adding-a-web-development-tool-to-safari-web-inspector?language=objc) with no change to behavior. I can't tell if this is an existing bug or not.

burg commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -276,18 +276,14 @@ void InspectorFrontendHost::inspectedURLChanged(const String& newURL)
-        if (RefPtr localMainFrame = m_frontendPage->localMainFrame())
-            return localMainFrame->pageZoomFactor();
-    }
+    if (m_client)
```
style: this whole thing could be one line now

burg commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -276,18 +276,14 @@ void InspectorFrontendHost::inspectedURLChanged(const String& newURL)
-        if (RefPtr localMainFrame = m_frontendPage->localMainFrame())
-            localMainFrame->setPageAndTextZoomFactors(zoom, 1);
-    }
+    if (m_client)
```
style: this whole thing could be one line now.

nipunshukla21 replied:
I'm not fully sure how to turn this one into a one-liner as making the if statement all go on one line violates the style checker.

nipunshukla21 committed:
Web Inspector: InspectorFrontendHost should not call setPageAndTextZoomFactors on the LocalFrame
https://bugs.webkit.org/show_bug.cgi?id=280493
rdar://136798696

Reviewed by BJ Burg.

Setting zoom for inspector previously used a function which recursively iterates through
the frame stack and sets state. However, we were using a variant which would not zoom
into RemoteFrames. This patch changes that to use the function from the UI Process,
which can traverse the full frame tree including local and remote frames.

* Source/WebCore/inspector/InspectorFrontendClient.h:
(WebCore::InspectorFrontendClient::setPageAndTextZoomFactors):
(WebCore::InspectorFrontendClient::pageZoomFactor const):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::setZoomFactor):
(WebCore::InspectorFrontendHost::zoomFactor):
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.cpp:
(WebKit::RemoteWebInspectorUIProxy::setPageAndTextZoomFactors):
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.h:
* Source/WebKit/UIProcess/Inspector/RemoteWebInspectorUIProxy.messages.in:
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.cpp:
(WebKit::WebInspectorUIProxy::setPageAndTextZoomFactors):
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.h:
* Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.messages.in:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::setPageAndTextZoomFactors):
(WebKit::RemoteWebInspectorUI::pageZoomFactor const):
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::setPageAndTextZoomFactors):
(WebKit::WebInspectorUI::pageZoomFactor const):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.h:

Canonical link: https://commits.webkit.org/301271@main
